src/js/components/history/HistoryItem.tsx

```diff
@@ -0,0 +1,192 @@
+import sg from '../../ctrl/signals';
+import {MODE, SELECTED, CELL} from '../../store/factory/Signal';
+import {NumericValueRef, StringValueRef, tupleid} from 'vega';
+import {channelName} from '../../actions/bindChannel';
+import {setMarkVisual} from '../../actions/markActions';
+import * as vega from 'vega';
+import bindChannel from '../../actions/bindChannel';
+import {ColumnRecord, Schema} from '../../store/factory/Dataset';
+import { AnyAction } from 'redux';
+import {ThunkDispatch} from 'redux-thunk';
+import {State} from '../../store';
+import {setSignal} from '../../actions/signalActions';
+import {SignalValue} from 'vega-typings/types';
+import {batchGroupBy} from '../../reducers/historyOptions';
+
+const ctrl = require('../../ctrl');
+
+
+interface OwnProps {
+  id: string,
+  history: HistoryRecord // TODO: use History.ts for you just have to pass the id
+
+  groupNames: any[];
+}
+interface DispatchProps {
+  startDragging: (d: DraggingStateRecord) => void;
+  stopDragging: () => void;
+  setMarkVisual: (payload: {property: string, def: NumericValueRef | StringValueRef}, markId: number) => void;
+
+  bindChannel: (dsId: number, field: ColumnRecord, markId: number, property: string) => void;
+
+  setSignal: (value: SignalValue, signal: string) => void;
+}
+
+function mapDispatchToProps(dispatch: ThunkDispatch<State, null, AnyAction>, ownProps: OwnProps): DispatchProps {
+  return {
+    startDragging: (d: DraggingStateRecord) => {
+      dispatch(startDragging(d)); },
+    stopDragging: () => {
+      dispatch(stopDragging()); },
+    setMarkVisual: (p, markId) => {
+      dispatch(setMarkVisual(p, markId));
+    },
+    bindChannel: (dsId: number, field: ColumnRecord, markId: number, property: string) => {
+      dispatch(bindChannel(dsId, field, markId, property));
+    },
+    setSignal: (value: SignalValue, signal: string) => {
+      dispatch(setSignal(value, signal));
+    }
+  };
+}
+
+export class HistoryItemInspector extends React.Component<OwnProps & DispatchProps> {
+
+  constructor(props) {
+    super(props);
+  }
+
+  private width = 100; // these should match baseSignals in demonstrations.ts
+  private height = 100; //
+
+  public handleClick = (historyId: number) => {
+
+  }
+  public handleDragStart = (historyId: number) => {
+    this.props.startDragging(HistoryDraggingState({historyId}));
+
+    sg.set(MODE, 'channels');
+    ctrl.update();
+  }
+
+  public handleDragEnd = (evt: React.DragEvent<HTMLDivElement>, opts?) => {
+    const props = this.props;
+    const sel = sg.get(SELECTED);
+    const cell = sg.get(CELL);
+    const dropped = tupleid(sel) && tupleid(cell);
+
+    try {
+      const lyraId = +sel.mark.role.split('lyra_')[1]; // id of thing that was dropped onto
+
+      if (dropped) {
+        const channel = channelName(cell.key);
+        let fieldName, dsId;
+        if (channel === 'x' || channel === 'y' || channel === 'color' || channel === 'size') {
```

Comment:
for other mark types, the channels won't always be `color` and `size`. i think if you're doing an if-case you should only case on `x` and `y`, and handle drops on the dropzones that are over the mark itself in the else case.

Reply:
yeah currently this is kinda aimed at symbols but will adapt for other mark types in PRs to follow